### PR TITLE
26 move regtech mail api under src directory

### LINKED ISSUE #26: Move regtech_mail_api under src directory
All python projects are being moved to a <repo>/src/<module> directory structure

Move the existing regtech_mail_api folder under a top level src directory.

Update any pathing needed.   For instance, Dockerfiles, pyproject.toml

Remove from pyproject.toml the depdency on fastapi. This will be pulled in with api-commons

### PR DESCRIPTION
**Pull request by jcadam14**
Closes #26 

This also includes the #25 changes since I was thinking that would have been approved by now but if not, can just encompass both.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -5,13 +5,12 @@ description = ""
 authors = ["CFPB's RegTech Team"]
 readme = "README.md"
 packages = [
-    { include = "regtech_mail_api" }
+    { include = "src/regtech_mail_api" }
```
Comment (hkeeler) [inline pyproject.toml:8]: We've done the other projects like...

```suggestion
    { include = "src/regtech_mail_api" }
    { include = "regtech_mail_api" from = "src"}
```

That way it knows the top-level module is in `src`, but it keep any additional files that're in `regtech_mail_api`, but not in `src`. Not an issue for this particular repo, but might as well keep it consistent.

**Review by hkeeler — COMMENTED** [review]
Overall looks good. Just the one tweak to `packages`.

File: pyproject.toml (hunk 1 above)
Comment (jcadam14) [reply pyproject.toml]: Consistency is good.  I saw that in the others but wasn't sure the difference between one vs the other.  Updated

**Comment by hkeeler** [comment]
Oops. Sorry. Gave a faulty example. Bad `,`!

**Comment by jcadam14** [comment]
> Oops. Sorry. Gave a faulty example. Bad `,`!

Ha I was looking at the poetry docs reading on the use of "from" and realized that right after I pushed.

**Review by hkeeler — APPROVED** [review]

**Pull request merged**